mla/models.py

```diff
@@ -290,209 +291,217 @@ def background_dec_spline(self) -> scipy.interpolate.UnivariateSpline:
                  data: np.ndarray,
                  sim: np.ndarray,
                  grl: np.ndarray,
-                 background_sin_dec_bins: Union[np.array, int] = 500,# Python 3.9 bug... pylint: disable=unsubscriptable-object
-                 signal_sin_dec_bins: Union[np.array, int] = 50,# Python 3.9 bug... pylint: disable=unsubscriptable-object
-                 log_energy_bins: Union[np.array, int] = 50,# Python 3.9 bug... pylint: disable=unsubscriptable-object
-                 spectrum: Optional[spectral.BaseSpectrum] = None,# Python 3.9 bug... pylint: disable=unsubscriptable-object
-                 sampling_width: Optional[float] = np.radians(3),# Python 3.9 bug... pylint: disable=unsubscriptable-object
-                 reduce_dec: Optional[bool] = True,# Python 3.9 bug... pylint: disable=unsubscriptable-object
-                 verbose: bool = False,
-                 **kwargs) -> None:
-        """Initializes EventModel and calculates background only signal-over-background (sob) maps for ThreeML.
-        
+                 background_sin_dec_bins: Union[np.array, int] = 500,  # Python 3.9 bug... pylint: disable=unsubscriptable-object
+                 signal_sin_dec_bins: Union[np.array, int] = 50,  # Python 3.9 bug... pylint: disable=unsubscriptable-object
+                 log_energy_bins: Union[np.array, int] = 50,  # Python 3.9 bug... pylint: disable=unsubscriptable-object
+                 spectrum: Optional[spectral.BaseSpectrum] = None,  # Python 3.9 bug... pylint: disable=unsubscriptable-object
+                 sampling_width: Optional[float] = np.radians(3),  # Python 3.9 bug... pylint: disable=unsubscriptable-object
+                 reduce_dec: Optional[bool] = True,  # Python 3.9 bug... pylint: disable=unsubscriptable-object
+                 verbose: bool = False) -> None:
+        """Initializes EventModel and calculates bg-only sob maps for ThreeML.
+
         More function info...
-        
+
         Args:
-            background_sin_dec_bins: If an int, then the number of bins spanning -1 -> 1,
-                otherwise, a numpy array of bin edges.
-            signal_sin_dec_bins: If an int, then the number of bins spanning -1 -> 1,
-                otherwise, a numpy array of bin edges.
+            background_sin_dec_bins: If an int, then the number of bins spanning
+                -1 -> 1, otherwise, a numpy array of bin edges.
+            signal_sin_dec_bins: If an int, then the number of bins spanning
+                -1 -> 1, otherwise, a numpy array of bin edges.
             log_energy_bins: if an int, then the number of bins spanning 1 -> 8,
                 otherwise, a numpy array of bin edges.
             spectrum: Spectrum for energy weighting
             reduce_dec: whether reduce the simulation to narrow dec
             verbose: A flag to print progress.
+
+        Raises:
+            ValueError:
         """
-        super(ThreeMLEventModel,self).__init__(data, sim, grl,
```

Comment:
What I want to do here is calling _create_log_sob_gamma_splines from ThreeMLEventModel when during __init__ which basically return None for _log_sob_gamma_splines . This is because the energy term for ThreeML is from get_energy_sob instead of splines. So I don't want to create the gamma splines in ThreeMLEventModel .